packages/loader/container-loader/src/container.ts

```diff
@@ -1251,17 +1254,29 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     }
 
     private get client(): IClient {
-        const client: IClient = this.options?.client !== undefined
-            ? (this.options.client as IClient)
-            : {
-                details: {
-                    capabilities: { interactive: true },
-                },
-                mode: "read", // default reconnection mode on lost connection / connection error
-                permission: [],
-                scopes: [],
-                user: { id: "" },
-            };
+        // augments the IClient type with the container session id.
```

Comment:
It seems a bit odd that ordering service would blindly put stuff it does not understand into file, but I guess it's side-effect of using JS / not ignoring unknown properties :)
This info already exists in telemetry, so I'm not sure duplicating it in the file adds value - it's a bit weird to add telemetry to ops.

Reply:
The difficutly is when analzing an op log, and trying to understand the sequence of events. We don't currently have a good way to combine op log, and telemetry data, and it's unclear to me if we will have telemetry data for all clients in the future.

Reply:
This could also allow new data integrity checks on the server or client to ensure there is only a single active connection/client for each container session id

Reply:
I think we are more likely to have telemetry than have access to a file.
I'd say that I agree with you it's useful to have that data on  IClient, but I'd rather make it official, i.e. change IClient definition itself. It might be useful data to have in user code, i.e. it appearing on Audience / Quorum - it's possible to see scenarios where we track a client (i.e. tab) and generate some report on user activity...

Reply:
Also if we have an official containerId, we should use the same one in the telemetry as well, so we can collate the result.

Reply:
The id i'm adding here is the container session id. if you requested two containers without caching i'd expect them both to be different. I don't think this should be container id. To me container id is more akin to what we call doc id, and represents the container and is constant regardless of the client or container instance, maybe something like this should be call the storage id.

Reply:
I think the better place for it is in IClient.details - it already tracks data about client itself:

export interface IClientDetails {
    capabilities: ICapabilities;
    type?: string;
    environment?: string;
    device?: string;
}